src/modules/AutoinstSoftware.rb

```diff
@@ -685,36 +685,38 @@ def createISO
     # @return dumped settings (later acceptable by Import())
     def Export
       s = {}
-      Ops.set(s, "kernel", @kernel) if @kernel != ""
+      s["kernel"] = @kernel if !@kernel.empty?
```

Comment:
`@kernel` and `@patterns` are initialized in the `main` method, so this call to `empty?` is safe.

Reply:
You could use `unless`, but that's up to you.

Reply:
yes it is initialized - otherwise I'd add nil check to. 

I'm still somehow old fashioned and prefer if over unless